fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/MvUtils.java

```diff
@@ -334,6 +334,7 @@ public static Pair<OptExpression, LogicalPlan> getRuleOptimizedLogicalPlan(Strin
                 new RelationTransformer(columnRefFactory, connectContext).transformWithSelectLimit(query);
         // optimize the sql by rule and disable rule based materialized view rewrite
         OptimizerConfig optimizerConfig = new OptimizerConfig(OptimizerConfig.OptimizerAlgorithm.RULE_BASED);
+        optimizerConfig.disableRuleSet(RuleSetType.PARTITION_PRUNE);
```

Comment:
This only affects MV's optimination? what about Query's `compensatePartitionPredicate`, query still need `compensatePartitionPredicate`?

Reply:
yes, query need to add total range to calculate compensation predicates